filesetup.c

```diff
@@ -1839,9 +1839,16 @@ void dup_files(struct thread_data *td, struct thread_data *org)
  */
 int get_fileno(struct thread_data *td, const char *fname)
 {
+	char file_name[PATH_MAX];
 	struct fio_file *f;
 	unsigned int i;
 
+	if (td->o.directory) {
+		int len = set_name_idx(file_name, PATH_MAX, td->o.directory, 0, false);
+		sprintf(file_name + len, "%s", fname);
```

Comment:
Any chance you can use snprintf? There are static analysis tools that start getting angsty when they can't prove you won't overflow the buffer...